Hello,

I tried with the latest version in develop (which includes #19963) and applied you commit for #20078.

But no luck when I try to install `spack install qt@5.14.2%gcc@8.3.0`

The patch in 19963 only applies to `gcc@10:` and I'm having the problem with gcc@8.3.0, so I'm trying to see what part of my original patch is still needed (definitely I need to patch `qjp2handler.cpp` but not sure about the others yet, since compilation stopped with the `strstring` error.  Not sure what is going on, because I can see that `doc` was not included:

```
angelv@sieladon:~$ spack spec  qt@5.14.2%gcc@8.3.0 | grep qt
==> Warning: gcc@8.3.0 cannot build optimized binaries for "cascadelake". Using best target possible: "skylake_avx512"
qt@5.14.2%gcc@8.3.0
qt@5.14.2%gcc@8.3.0~dbus~debug~doc~examples~framework~gtk+gui~opengl~phonon+shared+sql+ssl+tools~webkit patches=7f34d48d2faaa108dc3fcc47187af1ccd1d37ee0f931b42597b820f03a99864c,8449b9fba2e0a771a9f6b43f9aedcfb8c256e9945b86b54c2ac0484c352230bc arch=linux-arch-skylake_avx512

```

but it still seems to be needing llvm: 

```
             j/previewmanager.o shared/previewmanager.cpp                                                                                                                                                            
  >> 9133    /usr/bin/ld: /lib/../lib64/../lib/libLLVM-10.so: undefined reference to `std::_                                                                                                                         
             _cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >                                                                                                                         
             ::basic_stringstream()@@GLIBCXX_3.4.26'                                                                                                                                                                 
  >> 9134    collect2: error: ld returned 1 exit status                                                                                                                                                              
     9135    make[3]: *** [Makefile:232: ../../bin/qdoc] Error 1 
```

I will keep trying, but compiling Qt takes a lot of time...

--- a/var/spack/repos/builtin/packages/qt/package.py
+++ b/var/spack/repos/builtin/packages/qt/package.py
@@ -121,6 +121,7 @@ class Qt(Package):
     patch('qt5-12-configure.patch', when='@5.12')
     # https://bugreports.qt.io/browse/QTBUG-93402
     patch('qt5-15-gcc-10.patch', when='@5.12.7:5.15 %gcc@8:')
+    patch('qt514.patch', when='@5.14')
     conflicts('%gcc@10:', when='@5.9:5.12.6 +opengl')
 
     # Build-only dependencies

--- a/var/spack/repos/builtin/packages/qt/qt514.patch
+++ b/var/spack/repos/builtin/packages/qt/qt514.patch
@@ -1,0 +1,9 @@
+diff -ruN qt-everywhere-src-5.14.2.orig/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp qt-everywhere-src-5.14.2/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp
+--- qt-everywhere-src-5.14.2.orig/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp	2020-03-03 13:10:30.000000000 +0000
++++ qt-everywhere-src-5.14.2/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp	2020-11-16 19:57:56.792619032 +0000
+@@ -38,4 +38,5 @@
+ **
+ ****************************************************************************/
+ 
++#include <math.h>
+ #include "qjp2handler_p.h"

